### DISCUSSION TITLE
如何获取触发命令的消息id

### DISCUSSION THREAD

**Discussion by Xi-Lusheng** [nonebot/discussions#215] [问答] [answered]
![DI7UI77IV~L9RFJG ) IYZB](https://user-images.githubusercontent.com/93561701/194552904-b87edab2-b7a0-449f-9e33-c13bcacd1d99.png)
如图，我使用回复选择一条需要撤回的消息通过 `raw_message` 来获取消息的id撤回消息
但是由于qq的回复机制，依旧能在回复里面看见撤回的消息
有没有什么办法获得触发命令的消息id来同时撤回命令
或者其他方法获得需要撤回消息的id

**Comment by yanyongyu** [comment, marked as answer]
`event.message_id`应该是吧

回复的消息直接通过`event.reply`获取就行，不用`raw_message`

**Reply by Xi-Lusheng** [reply]
ok，感谢
